src/main/java/org/mybatis/spring/MyBatisExceptionTranslator.java

```diff
@@ -94,7 +98,7 @@ public DataAccessException translateExceptionIfPossible(RuntimeException e) {
       } else if (e.getCause() instanceof TransactionException) {
         throw (TransactionException) e.getCause();
       }
-      return new MyBatisSystemException(e);
+      return new MyBatisSystemException(msg, e);
```

Comment:
I suggest to add new assertion for `msg` at `SqlSessionTemplateTest#testExceptionTranslationShouldThrowMyBatisSystemException`.